src/proof/CvcMerkleProof/Ed25519SignerVerifier.ts

```
@@ -0,0 +1,59 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import nacl from "tweetnacl";
+import {MerkleProof} from "../../lib/signerVerifier";
+import {SignerVerifier} from "./SignerVerifier";
+import {IDiDResolver} from "../../lib/resolver";
+import {Keypair} from "@solana/web3.js";
+import {VerifiableCredential} from "../../vc/VerifiableCredential";
+import didUtil from "../../lib/did";
+import bs58 from "bs58";
+
+export class Ed25519SignerVerifier implements SignerVerifier {
+    constructor(private _didResolver: IDiDResolver, private _verificationMethod: string, private _keypair: Keypair) {
+    }
+
+    async sign(proof: MerkleProof): Promise<any> {
+        if (!this._keypair) throw new Error(`No private key provided for signing`);
+
+        const signed = nacl.sign.detached(Buffer.from(proof.merkleRoot, 'hex'), this._keypair.secretKey);
+        const signature = Buffer.from(signed).toString('hex');
+
+        return Promise.resolve({
+            signature,
+            verificationMethod: this._verificationMethod,
+        });
+    }
+
+    async verify(vc: VerifiableCredential): Promise<boolean> {
+        const did = await didUtil.resolve(vc.issuer, this._didResolver);
+        if (!did) throw new Error(`Unable to resolve document for ${vc.issuer}`);
+
+        const vm = await didUtil.findVerificationMethod(did, this._verificationMethod);
+
+        if(!vm) return false;
```

Should we return an error?

---

No... it was like that... but, following suite on other libraries, if it is unable to verify (for whatever reason) return false